tests/func/test_repro_multistage.py

```diff
@@ -518,3 +515,44 @@ def test_repro_multiple_params(tmp_dir, dvc):
     dump_yaml(tmp_dir / "params.yaml", params)
 
     assert dvc.reproduce(stage.addressing) == [stage]
+
+
+def test_repro_list_of_commands_in_order(tmp_dir, dvc):
+    dvcfile_content = {
+        "stages": {
+            "multi_commands": {
+                "cmd": [
+                    """echo "First command" > foo""",
+                    """echo "Second command" >> foo""",
+                ],
+                "outs": ["foo"],
+            }
+        }
+    }
+    dump_yaml(PIPELINE_FILE, dvcfile_content)
+    dvc.reproduce(target="multi_commands")
+    with open(Path(tmp_dir) / "foo", "r") as foo:
+        lines = foo.read().splitlines()
+    assert lines == ["First command", "Second command"]
+
+
+def test_repro_list_of_commands_raise_and_stops_after_failure(tmp_dir, dvc):
+    dvcfile_content = {
+        "stages": {
+            "multi_commands": {
+                "cmd": [
+                    """echo "First command" > foo""",
+                    """ehco "Second command" >> foo""",
```

Comment:
Lol, took me a second to realize what's going on :smile: Could make this a bit more obvious, but up to you.

Reply:
changes `"""` to `'` hopefully it is clearer